forest/benchmarking/distance_measures.py

```diff
@@ -33,10 +35,31 @@ def purity(rho, dim_renorm=False):
         P = (dim / (dim - 1.0)) * (Ptemp - 1.0 / dim)
     else:
         P = np.trace(np.matmul(rho, rho))
```

Comment:
This might be more clear if we calculate the purity, then renormalize if it is necessary.
```python
p = np.trace(rho @ rho)
if dim_renorm:
    dim = rho.shape[0]
    p = (dim / (dim - 1.0)) * (p - 1.0 / dim)
```

Reply:
- [x] todo